app/server/lib/ConfigBackendAPI.ts

```diff
@@ -0,0 +1,35 @@
+import * as express from 'express';
+import {expressWrap} from 'app/server/lib/expressWrap';
+
+import {IGristCoreConfig, loadGristCoreConfigFile} from 'app/server/lib/configCore';
+import log from "app/server/lib/log";
+
+export class ConfigBackendAPI {
+  private _config: IGristCoreConfig;
+  public async init () {
+      this._config = await loadGristCoreConfigFile();
```

Comment:
Use this instead of loading the core config file directly:
https://github.com/gristlabs/grist-core/blob/main/stubs/app/server/lib/globalConfig.ts

There's a potential race condition otherwise if multiple configs in different states exist. (Unfortunate quirk of being able to only write a whole JSON file)

Reply:
You're also not passing a path to this - which means it's using an in-memory store rather than a file-backed one :)

Reply:
I tried reading the code and thought it would default to a file-based config. It even emits log messages that it's reading the file path.